### PR TITLE
ci: use hashes when installing pip dependencies

### PR DESCRIPTION
**Pull request by coryan**
This is only required for dependencies that publish packages or documentations.  Packages used in tests are excluded.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9760)

### CODE REVIEW

File: ci/cloudbuild/builds/publish-docs.sh [hunk 1]
```diff
@@ -73,9 +73,22 @@ if [[ "${PROJECT_ID:-}" != "cloud-cpp-testing-resources" ]]; then
...
+#    pip freeze | grep -v gcp-docuploader >ci/docuploader-requirements.txt
+#    pip install hashin
+#    for package in $(sed -n '/==/ s/\(.*\)==.*/\1/p' ci/docuploader-requirements.txt); do
+#      hashin "${package}" -r ci/docuploader-requirements.txt;
+#    done
+# Apologies, we can try to simplify this later.
 io::log_h2 "Installing the docuploader package"
 python3 -m pip install --upgrade --user --quiet --disable-pip-version-check \
-  --no-warn-script-location \
+  --no-warn-script-location --require-hashes -r ci/docuploader-requirements.txt
```
Comment (devbww) [inline ci/cloudbuild/builds/publish-docs.sh:89]: Perhaps out it in `ci/etc/`?

File: ci/cloudbuild/builds/publish-docs.sh [hunk 2]
```diff
@@ -73,9 +73,22 @@ if [[ "${PROJECT_ID:-}" != "cloud-cpp-testing-resources" ]]; then
   exit 0
 fi
 
+# See b/242562806 for the motivation. To update ci/docuploader-requirements.txt you need to do something like:
```
Comment (devbww) [inline ci/cloudbuild/builds/publish-docs.sh:76]: Does "b/..." have any currency in OSS land?

File: ci/cloudbuild/builds/publish-docs.sh (hunk 1 above)
Comment (coryan) [reply ci/cloudbuild/builds/publish-docs.sh]: Done.

File: ci/cloudbuild/builds/publish-docs.sh (hunk 2 above)
Comment (coryan) [reply ci/cloudbuild/builds/publish-docs.sh]: I hope it is clearer now.

**Review by devbww — APPROVED** [review]

**Pull request merged**